geninfo: ERROR: Unexpected negative count '-11' for /home/runner/work/skupper-router/skupper-router/skupper-router/tests/alloc_test.c:130.

This often happens on fedora 39

```
Processing /__w/skupper-router/skupper-router/skupper-router/build/tests/CMakeFiles/unit_tests.dir/alloc_test.c.gcda
geninfo: ERROR: Unexpected negative count '-11' for /home/runner/work/skupper-router/skupper-router/skupper-router/tests/alloc_test.c:130.
	Perhaps you need to compile with '-fprofile-update=atomic
	(use "geninfo --ignore-errors negative ..." to bypass this error)
Combining tracefiles.
.. found 2 files to aggregate.
Merging dispatch-base.info..1 remaining
Merging dispatch-ctest.info..0 remaining
lcov: ERROR: trace file 'dispatch-ctest.info' is empty
	(use "lcov --ignore-errors empty ..." to bypass this error)
lcov: ERROR: no valid records found in tracefile dispatch-total-raw.info
	(use "lcov --ignore-errors empty ..." to bypass this error)
genhtml: ERROR: no valid records found in tracefile dispatch-total.info
	(use "genhtml --ignore-errors empty ..." to bypass this error)
gmake[3]: *** [CMakeFiles/coverage.dir/build.make:70: CMakeFiles/coverage] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:992: CMakeFiles/coverage.dir/all] Error 2
gmake[3]: Leaving directory '/__w/skupper-router/skupper-router/skupper-router/build'
gmake[2]: Leaving directory '/__w/skupper-router/skupper-router/skupper-router/build'
gmake[1]: *** [CMakeFiles/Makefile2:999: CMakeFiles/coverage.dir/rule] Error 2
gmake[1]: Leaving directory '/__w/skupper-router/skupper-router/skupper-router/build'
gmake: *** [Makefile:543: coverage] Error 2
```

https://github.com/skupperproject/skupper-router/actions/runs/7340178454/job/19985745701#step:35:1417

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -307,15 +307,15 @@ endif()
 # Set up extra coverage analysis options for gcc and clang
 if (CMAKE_BUILD_TYPE MATCHES "Coverage")
  if (ENABLE_PROFILE_GUIDED_OPTIMIZATION)
-   set (CMAKE_C_FLAGS_COVERAGE "-g -O2 -DNDEBUG --coverage")
-   set (CMAKE_CXX_FLAGS_COVERAGE "-g -O2  --coverage")
+   set (CMAKE_C_FLAGS_COVERAGE "-g -O2 -DNDEBUG --coverage -fprofile-update=atomic")
+   set (CMAKE_CXX_FLAGS_COVERAGE "-g -O2  --coverage -fprofile-update=atomic")
  else(ENABLE_PROFILE_GUIDED_OPTIMIZATION)
-   set (CMAKE_C_FLAGS_COVERAGE "-g -O0 --coverage")
-   set (CMAKE_CXX_FLAGS_COVERAGE "-g -O0 --coverage")
+   set (CMAKE_C_FLAGS_COVERAGE "-g -O0 --coverage -fprofile-update=atomic")
+   set (CMAKE_CXX_FLAGS_COVERAGE "-g -O0 --coverage -fprofile-update=atomic")
  endif(ENABLE_PROFILE_GUIDED_OPTIMIZATION)
- set (CMAKE_EXE_LINKER_FLAGS_COVERAGE "--coverage")
- set (CMAKE_MODULE_LINKER_FLAGS_COVERAGE "--coverage")
- set (CMAKE_SHARED_LINKER_FLAGS_COVERAGE "--coverage")
+ set (CMAKE_EXE_LINKER_FLAGS_COVERAGE "--coverage -fprofile-update=atomic")
+ set (CMAKE_MODULE_LINKER_FLAGS_COVERAGE "--coverage -fprofile-update=atomic")
+ set (CMAKE_SHARED_LINKER_FLAGS_COVERAGE "--coverage -fprofile-update=atomic")
 
  mark_as_advanced(
    CMAKE_C_FLAGS_COVERAGE

